src/test/java/ui/WeekViewTest.java

```diff
@@ -0,0 +1,59 @@
+package ui;
+
+import data.TaskManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import time.WeekView;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class WeekViewTest {
+
+    private WeekView weekView;
+    private TaskManager taskManagerMock;
+
+    @BeforeEach
+    public void setUp() {
+        LocalDate startOfWeek = LocalDate.of(2024, 3, 10); // March 10, 2024
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("MM/dd/yyyy");
+        weekView = new WeekView(startOfWeek, dateFormatter);
+        taskManagerMock = new TaskManager(); // You can create a mock TaskManager as needed
+    }
+
+    @Test
+    public void testNextWeek() {
```

Comment:
Likewise for testNextWeek and testPreviousWeek it could perhaps be refactored into one method